app/server/identity/identityMiddleware.ts

```diff
@@ -33,6 +33,47 @@ const filterQueryParametersByName = (
     .reduce((params2, [name, value]) => ({ ...params2, [name]: value }), {});
 };
 
+// Names of query parameters to that facilitate sign-in on profile.
+const signInTokenQueryParameterNames = ["encryptedEmail", "autoSignInToken"];
+
+const containsSignInTokenQueryParameters = (
+  req: MockableExpressRequest
+): boolean =>
+  signInTokenQueryParameterNames.some(name => req.query[name] !== undefined);
+
+// Adds the redirect url (if defined) as query parameter profileReferer,
+// and removes the sign-in token query parameters since they are not required by manage
+// (only used by profile if the user is redirected their to sign-in).
+const updateManageUrl = (
+  req: MockableExpressRequest,
+  useRefererHeader: boolean,
+  redirectUrl?: UrlWithParsedQuery
+): string => {
+  // It is vital that the sign-in query parameters are removed.
+  // See the implementation of withIdentity() for more context.
+  const queryParameters = filterQueryParametersByName(
+    req.query,
+    name => !signInTokenQueryParameterNames.includes(name)
+  );
+
+  const profileReferrer =
+    redirectUrl && redirectUrl.path ? redirectUrl.path.substring(1) : undefined;
```

Comment:
I don't know the precedence of operators off the top of my head, so I'd have to look that up if I wanted to understand what this did. Is there another way of expressing this or just putting extra braces in?

Reply:
`yarn fix` removes the brackets, so will keep as is